utils/lazy-pages-fuzzer/src/generate/globals.rs

```diff
@@ -0,0 +1,129 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use anyhow::Result;
+use arbitrary::{Arbitrary, Unstructured};
+
+use gear_wasm_instrument::parity_wasm::{
+    builder,
+    elements::{Instruction, Module},
+};
+
+pub const GLOBAL_NAME_PREFIX: &str = "gear_fuzz_";
+
+#[derive(Debug, Clone)]
+pub struct InjectGlobalsConfig {
+    pub max_global_number: usize,
+    pub max_access_per_func: usize,
+}
+
+impl Default for InjectGlobalsConfig {
+    fn default() -> Self {
+        InjectGlobalsConfig {
+            max_global_number: 10,
+            max_access_per_func: 10,
+        }
+    }
+}
+
+pub struct InjectGlobals<'u> {
+    unstructured: Unstructured<'u>,
+    config: InjectGlobalsConfig,
+}
+
+impl InjectGlobals<'_> {
+    pub fn new(unstructured: Unstructured<'_>, config: InjectGlobalsConfig) -> InjectGlobals<'_> {
+        InjectGlobals {
+            unstructured,
+            config,
+        }
+    }
+
+    pub fn inject<'this>(mut self, mut module: Module) -> Result<(Module, Unstructured<'this>)>
```

Comment:
I'd introduce a unit test for that fn checking that globals are injected propertly, and access instruction actually access expected globals

Reply:
Added test